Sources/Workspace/Workspace.swift

```diff
@@ -980,25 +964,30 @@ public class Workspace {
     private func validateEditedPackages() throws {
         let managedDependencies = try self.managedDependencies.load()
         for dependency in managedDependencies.values {
-
-            let dependencyPath: AbsolutePath
-
             switch dependency.state {
             case .checkout: continue
             case .edited:
-                dependencyPath = editablesPath.appending(dependency.subpath)
-            case .unmanaged(let path):
-                dependencyPath = path
-            }
-
-            // If some edited dependency has been removed, mark it as unedited.
-            if !fileSystem.exists(dependencyPath) {
-                try unedit(dependency: dependency, forceRemove: true)
-                // FIXME: Use diagnosics engine when we have that.
-                delegate.warning(message: "\(dependencyPath.asString) was being edited but has been removed, falling back to original checkout.")
+                let dependencyPath = path(for: dependency)
+                
+                // If some edited dependency has been removed, mark it as unedited.
+                if !fileSystem.exists(dependencyPath) {
+                    try unedit(dependency: dependency, forceRemove: true)
+                    // FIXME: Use diagnosics engine when we have that.
+                    delegate.warning(message: "\(dependency.subpath.asString) was being edited but has been removed, falling back to original checkout.")
+                }
             }
         }
     }
+    
+    /// Returns the location of the dependency
+    private func path(for dependency: ManagedDependency) -> AbsolutePath {
```

Comment:
`compute` doesn't add much information though. Btw, `FileManager` now has similar named functions since Swift 3:

```
func url(for directory: FileManager.SearchPathDirectory, in domain: FileManager.SearchPathDomainMask, appropriateFor url: URL?, create shouldCreate: Bool) throws -> URL
```